### PR TITLE
Add a clean rustspec.cc

### PR DESCRIPTION
**Pull request by NalaGinrut**
I found the current `rustspec.cc` is directly copied from gccgo, this may cause some unwilling issues because go and rust are different languages. 
This patch provides a clean and minimum `rustspec.cc`. I think we can add more things to it when it's necessary. 

BTW, I found there're other files copied from gccgo, I may try to clean them to make sure they work smoothly for us.

PS: could you authorize me to push branches? Thanks!

### CODE REVIEW

**Comment by NalaGinrut** [comment]
There's another issue that `insn-modes.h` can't be found:
```
In file included from ../../gcc/rust/rustspec.cc:22:
../../gcc/coretypes.h:424:10: fatal error: insn-modes.h: No such file or directory
 #include "insn-modes.h"
          ^~~~~~~~~~~~~~
```
In the beginning, I thought it's because of the unclean `rustspec.cc`, however, this file exists in $(BUILD_DIR)/gcc. So it's another unrelavent issue. I may send another PR for it.

**Comment by NalaGinrut** [comment]
Interesting, in this case, `insn-modes.h` can only be generated under `make` rather than `make -j n`. So that this compiling error wasn't caught by CI since the script doesn't use `make -j n`.

**Comment by SimplyTheOther** [comment]
> Interesting, in this case, `insn-modes.h` can only be generated under `make` rather than `make -j n`. So that this compiling error wasn't caught by CI since the script doesn't use `make -j n`.

I've also noticed this issue - I get around it by running `make -j 4` and then after the compilation fails, I run `make`, and then I suspend that compilation after it produces `insn-modes.h` and then rerun `make -j 4` again. I assumed it was a dependency ordering failure issue rather than not being produced at all, but I could be wrong.



> I found the current `rustspec.cc` is directly copied from gccgo, this may cause some unwilling issues because go and rust are different languages.
> This patch provides a clean and minimum `rustspec.cc`. I think we can add more things to it when it's necessary.
> 
> BTW, I found there're other files copied from gccgo, I may try to clean them to make sure they work smoothly for us.
> 
> PS: could you authorize me to push branches? Thanks!

Thanks for the enthusiasm. However, as philberty (the original gccrs author) has merged my changes into his master, and as I consider his the "authoritative version", I'm currently treating this repository as my personal "working branch" - I'd suggest opening a PR at the upstream philberty/gccrs. He merged mine within a day of opening it, so I'm sure he won't take too long to merge yours (unless he has objections to it). I actually had a "clean" `rustspec.cc` similar to yours in my repo here before philberty rebased his on top of mine (which I then switched to working on - you can see the old master in the old-pre-mainline branch), but philberty replaced it for some reason with his copied gccgo one. I'm not sure whether he had a specific reason for doing so or if it just slipped past, but he may actually have reasons for doing so that I don't know about. 

Anyway TL;DR is that I'd suggest opening this PR on philberty/gccrs instead as that seems to be the main gccrs repository for the time being - this one is just for creating my PRs from.

**Comment by NalaGinrut** [comment]
OK, thanks for the reply! I'll move back to `philberty/gccrs`
